src/main/java/io/strimzi/kproxy/internal/interceptor/DefaultHandlerContext.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.strimzi.kproxy.internal.interceptor;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.util.Attribute;
+import io.strimzi.kproxy.interceptor.HandlerContext;
+import io.strimzi.kproxy.internal.AttributeKeys;
+
+public class DefaultHandlerContext implements HandlerContext {
+
+    private final ChannelHandlerContext ctx;
+
+    public DefaultHandlerContext(ChannelHandlerContext ctx) {
+        this.ctx = ctx;
+    }
+
+    @Override
+    public String channelDescriptor() {
+        return ctx.channel().toString();
+    }
+
+    @Override
+    public ByteBuffer allocate(int initialCapacity) {
```

Comment:
I would expose a `ByteBuf` as suggested in https://github.com/gunnarmorling/kproxy/pull/1:
- if the Netty `ByteBuf` change capacity, this `ByteBuffer` instance won't be available anymore and referencing it will cause a JVM segmentation fault or worst -> if reused will cause writing on data owned elsewhere!)